src/ast/codegen_llvm.cpp

```diff
@@ -507,7 +507,7 @@ void CodegenLLVM::visit(Call &call)
     Value *inet_offset = b_.CreateGEP(buf, {b_.getInt64(0), b_.getInt64(4)});
 
     inet->accept(*this);
-    if (inet->type.type == Type::array) {
+    if (inet->type.type == Type::array || inet->type.type == Type::string) {
       b_.CreateProbeRead(reinterpret_cast<AllocaInst *>(inet_offset), inet->type.size, expr_);
```

Comment:
Just spotted this neighbouring code - it should be a `static_cast` not `reinterpret_cast`, since `AllocaInst` inherits from `Value`